fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -595,6 +594,15 @@ private OptExpression logicalRuleRewrite(
         return tree.getInputs().get(0);
     }
 
+    private void rewriteGroupingSets(OptExpression tree, TaskContext rootTaskContext, SessionVariable sessionVariable) {
+        if (sessionVariable.isCboPushDownGroupingSet()) {
+            ruleRewriteOnlyOnce(tree, rootTaskContext, new PushDownAggregateGroupingSetsRule());
+        }
+        if (sessionVariable.isEnableRewriteGroupingsetsToUnionAll()) {
+            ruleRewriteIterative(tree, rootTaskContext, new RewriteGroupingSetsByCTERule());
+        }
+    }
```

Comment:
If both rules are turn on, will it split to union twice?

Reply:
emmm, you are right, will execute two rules and split all to cte in final, may be we need execute `RewriteGroupingSetsByCTERule` frist